### ISSUE TITLE
Fork and GCM

### ISSUE THREAD

**Issue by dunajski** [ForkIssues/TrackerWin#669]
Few of my workmates have problem with credentials.
Occasionally, when they fetch pop-up Windows GCM to login to repository.
We found out that the problem can be with login and `user.name` especially after Fork update.
In our situation, this is private repo and we use diffrent login and `user.name` (that's normal), where login looks NameSurname and then `user.name` looks Name Surname.
When problem occurs changing, Windows GCM pop-up and we need to login again and then our `user.name`s change to login. I couldn't find more. Is there any solution for that?
I've to admit this is not happening for everyone.

How Paddle-Fork and git: credential works in Windows GCM?

**Comment by DanPristupov** [comment]
/just updated with more details/

Fork doesn't change `user.name` without user request.

> When problem occurs changing, Windows GCM pop-up and we need to login again and then our user.names change to login.

Do you use Kraken? I received many reports that some application changed `user.name` silently and it turned out to be Git Kraken.

> When problem occurs changing, Windows GCM pop-up and we need to login again and then our user.names change to login. I couldn't find more. Is there any solution for that?

GCM shouldn't touch git config.

> How Paddle-Fork and git: credential works in Windows GCM?

Paddle is definitely not related.

**Comment by dunajski** [comment]
> Do you use Kraken? I received many reports that some application changed user.name silently and it turned out to be Git Kraken.

No.

> GCM shouldn't touch git config.

I know but it looks like somehow Fork had lost credentials and when  fetch gcm window poped-up to log in again and then it changed `user.name` to version without space between, NameSurname.

**Comment by DanPristupov** [comment]
Are you sure you see the GCM window? Open Terminal (Ctrl+Alt+T) and run ```git config --system credential.helper manager``` to be sure.

I just checked the source code.

Fork changes **global** credentials in two cases:
- In Welcome Dialog after the first install
- When username in Fork Preferences dialog (Ctrl+,) is **modified** by user

Fork changes **local** username in one case:
- When username in Repository Preferences dialog is **modified** by user


Try to think what other tools can be involved. IDE, hooks?

**Comment by dunajski** [comment]
yeah but, there is a connection between login and `user.name`? I think it shouldn't.
I'll look for what else can "mix" credentials.

**Comment by DanPristupov** [comment]
There is no connection between authentication and the `user.name`.

> I'll look for what else can "mix" credentials.

First find out what particular file contain those changes:
```$ git config --list --show-origin```

Then check the modification time of this file. Try to recall what happened at this time.
